### PR TITLE
Core/Characters: Added allied race creation data, implemented intro s…

### PR DESCRIPTION
**Pull request by mdX7**
…cenes & added config option to disable achievement requirements for allied races

**Changes proposed:**

-  Added playercreateinfo data for allied races
-  Added config option to disable required achievements for allied races
-  Implemented scene intros

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**

(Does it build, tested in-game, etc.)
Builds, tested ingame

**Depends on:** (add/remove lines as needed)
#26975 to fix blackscreens after skipping intro

### CODE REVIEW

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -910,6 +910,13 @@ StartAlliedRacePlayerLevel = 10
 
 StartPlayerMoney = 0
 
+#
+#    DisableAlliedRaceAchievementRequirement
+#        Description: Disable achievement requirements for allied race character creation
+#        Default:     0 (Keep requirements active)
+#                     1 (Disable requirements)
+DisableAlliedRaceAchievementRequirement = 0
```
Comment (Jinnaix) [inline src/server/worldserver/worldserver.conf.dist:918]: My OCD kicks in - please add a blank line above here :D
```suggestion

DisableAlliedRaceAchievementRequirement = 0
```

**Comment by Shauren** [comment]
What about NPE scene?

**Comment by MaxtorCoder** [comment]
NPE scene is different, no idea if it can be handled generically.

**Comment by mdX7** [comment]
> 
> 
> NPE scene is different, no idea if it can be handled generically.

very likely that it can be handled using the same method, will look into it today

**Comment by MaxtorCoder** [comment]
> > NPE scene is different, no idea if it can be handled generically.
> 
> very likely that it can be handled using the same method, will look into it today

👀👀👀

**Comment by mdX7** [comment]
works in a way, feels like wrong map is used in scene play - could it be that the map for map is selected by transport guid? for reference:
Retail:
```
ServerToClient: SMSG_PLAY_SCENE (0x262B) Length: 45 ConnIdx: 1 Time: 10/04/2021 08:09:30.656 Number: 281
SceneID: 2236
PlaybackFlags: 40
SceneInstanceID: 1
SceneScriptPackageID: 2578
TransportGUID: Full: 0x1806B080000000000001856871100001 Transport/0 R428/S389 Map: 1024 (Greenstone Village) Low: 448573472769
Pos: X: 11.1301 Y: -0.417182 Z: 5.18741
Facing: 3.148434400558471679
Encrypted: False
```
TC with this PR:
```
ServerToClient: SMSG_PLAY_SCENE (0x262B) Length: 38 ConnIdx: 1 EP: 127.0.0.1:64168 Time: 10/04/2021 08:48:21.898 Number: 180
SceneID: 2236
PlaybackFlags: 40
SceneInstanceID: 1
SceneScriptPackageID: 2578
TransportGUID: Full: 0x18000740000000000000000000000000 Transport/0 R1/S0 Map: 6656 Low: 0
Pos: X: 11.1301 Y: -0.417182 Z: 5.18741
Facing: 3.148430109024047851
Encrypted: False
```

**Comment by MaxtorCoder** [comment]
> 
> 
> works in a way, feels like wrong map is used in scene play - could it be that the map is selected by transport guid? for reference: Retail:
> 
> ```
> ServerToClient: SMSG_PLAY_SCENE (0x262B) Length: 45 ConnIdx: 1 Time: 10/04/2021 08:09:30.656 Number: 281
> SceneID: 2236
> PlaybackFlags: 40
> SceneInstanceID: 1
> SceneScriptPackageID: 2578
> TransportGUID: Full: 0x1806B080000000000001856871100001 Transport/0 R428/S389 Map: 1024 (Greenstone Village) Low: 448573472769
> Pos: X: 11.1301 Y: -0.417182 Z: 5.18741
> Facing: 3.148434400558471679
> Encrypted: False
> ```
> 
> TC with this PR:
> 
> ```
> ServerToClient: SMSG_PLAY_SCENE (0x262B) Length: 38 ConnIdx: 1 EP: 127.0.0.1:64168 Time: 10/04/2021 08:48:21.898 Number: 180
> SceneID: 2236
> PlaybackFlags: 40
> SceneInstanceID: 1
> SceneScriptPackageID: 2578
> TransportGUID: Full: 0x18000740000000000000000000000000 Transport/0 R1/S0 Map: 6656 Low: 0
> Pos: X: 11.1301 Y: -0.417182 Z: 5.18741
> Facing: 3.148430109024047851
> Encrypted: False
> ```

Feels more like the transport guid is constructed incorrectly.

**Comment by mdX7** [comment]
but even then, mapid 1024 isn't NPE

**Comment by MaxtorCoder** [comment]
> 
> 
> but even then, mapid 1024 isn't NPE

Because Transport guid is constructed incorrectly, I remember seeing something similar with my code in NPE.

**Review by Shauren — APPROVED** [review]

**Comment by uipoet** [comment]
@mdX7 You are a hero among pull requesters! My ten-year-old daughter is thrilled to play the allied races she has seen grayed-out on our home fun server for some time. Cheers!

**Pull request merged**